assets/blocks/learner-courses-block/learner-courses-edit.js

```diff
@@ -1,23 +1,153 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
 /**
  * Internal dependencies
  */
+import CourseProgress from './course-progress';
 import LearnerCoursesSettings from './learner-courses-settings';
 
 /**
  * Learner Settings component.
  *
  * @param {Object}   props
+ * @param {Object}   props.className          Block className.
  * @param {Object}   props.attributes         Block attributes.
  * @param {Object}   props.attributes.options Block options attribute.
  * @param {Function} props.setAttributes      Block set attributes function.
  */
-const LearnerCoursesEdit = ( { attributes: { options }, setAttributes } ) => {
+const LearnerCoursesEdit = ( {
+	className,
+	attributes: { options },
+	setAttributes,
+} ) => {
+	const [ filter, setFilter ] = useState( 'all' );
+
+	const filterHandler = ( filterValue ) => ( e ) => {
+		e.preventDefault();
+
+		setFilter( filterValue );
+	};
+
+	const filters = [
+		{
+			label: __( 'All courses', 'sensei-lms' ),
+			value: 'all',
+		},
+		{
+			label: __( 'Active courses', 'sensei-lms' ),
+			value: 'active',
+		},
+		{
+			label: __( 'Completed courses', 'sensei-lms' ),
+			value: 'completed',
+		},
+	];
+
+	// Set options function used for block settings.
 	const setOptions = ( editedOptions ) =>
 		setAttributes( { options: { ...options, ...editedOptions } } );
 
+	// Courses placeholder map function.
+	const coursesPlaceholderMap = ( i, index, array ) => {
+		const completed =
+			// All items should be in progress if active filter is selected.
+			filter !== 'active' &&
+			//  Show last one as completed.
+			( index === array.length - 1 ||
+				//  Show all as completed if completed is filtered.
+				filter === 'completed' );
+
+		return (
+			<li
+				className="wp-block-sensei-lms-learner-courses__courses-list__item"
+				key={ index }
+			>
+				{ options.featuredImageEnabled && (
+					<div
+						className="wp-block-sensei-lms-learner-courses__courses-list__featured-image"
+						role="img"
+						aria-label="Featured image"
```

Comment:
When rendering in the frontend with the image, we should just add a `background-image` in this `div`, so with the `backgrond-size: cover` it'll work fine for all variations. Also concat the course title in the `aria-label` could be a good idea.